src/coreclr/vm/threadsuspend.cpp

```diff
@@ -6077,7 +6076,7 @@ void ThreadSuspend::Initialize()
             }
```

Comment:
We can assert that `g_returnAddressHijackTarget` is non-null here. If `g_returnAddressHijackTarget` is NULL and shadow stacks are enabled, bad things will happen later.

Reply:
It would be an OS bug or misuse of the API, but we had a good deal of unexpected behaviors in the past with various legacy/beta OS versions, emulators, WOW, etc...

Assert would make sense.